### ISSUE TITLE
1 isoforms were only found in the annotation

### ISSUE THREAD

**Issue by MetteBoge** [kvittingseerup/IsoformSwitchAnalyzeR#257]
I try to load in data from TCGA using importRdata: 
```
switchAnalyzeRlist <- IsoformSwitchAnalyzeR::importRdata(isoformCountMatrix = transcript_count_ctype,
            isoformRepExpression = transcript_tpm_ctype,
            designMatrix = myDesign_ctype,
            isoformExonAnnoation = file_path_gtf
            )
```
But in step 2 I get this warning: 
```
Step 2 of 10: Obtaining annotation...
    importing GTF (this may take a while)...
Advarsel i IsoformSwitchAnalyzeR::importRdata(isoformCountMatrix = transcript_count_ctype,  :
  The annotation and quantification (count/abundance matrix and isoform annotation) Seem to be slightly different. 
Specifically:
 1 isoforms were only found in the annotation
 
Please make sure this is on purpouse since differences will cause inaccurate quantification and thereby skew all analysis.
 If you have quantified with Salmon this could be normal since it as default only keep one copy of identical sequnces (can be prevented using the --keepDuplicates option)
 We strongly encurage you to go back and figure out why this is the case.


    2 ( 3.12%) isoforms were removed since they were not expressed in any samples.

```

My gtf file is downloaded from this link: https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_23/gencode.v23.annotation.gtf.gz
Which should be the same used for generating the tpm and count files. 

I do not understand why it only finds 1 isoform in the annotation? If I unpack the gtf file and search for the transcript IDs I am able to find them!

These two expressions are TRUE for all: 
```
colnames(transcript_tpm_ctype) == colnames(transcript_count_ctype)
transcript_tpm_ctype$isoform_id == transcript_count_ctype$isoform_id
```

Here is a little peak of what my two data frames look like: 
<img width="255" alt="image" src="https://github.com/user-attachments/assets/a7ac162e-6b1b-4d9f-b457-67c96c265a1d">

Because of this warning, I get a `switchAnalyzeRlist` looking like: 
![image](https://github.com/user-attachments/assets/16c3ac6d-f440-4f0f-843a-5d35cbfb9ebe)

Hope you can help debugging this issue. 
Best regards, 
Mette

**Comment by chunxubioinfor** [comment]
Hi Mette, that looks really weird. Can you use the importGTF() function to import the GTF file downloaded, and then check if the isoform_id column matches the isoform_id in your count matrix? You could also send a subset of the data. frames (like the top 100 rows) and I can try to figure out what's going on. :)

**Comment by MetteBoge** [comment]
## GTF file: 
```
> aSwitchList <- importGTF(pathToGTF = file_path_gtf)
> intersect(aSwitchList[[1]]$isoform_id, transcript_tpm_ctype$isoform_id) %>% length()
[1] 64
```
I have a total of 64 isoforms in the count matrix, so it looks like they are all included in the GTF file. 

I attach my tpm and count files.
[transcript_tpm_ctype.csv](https://github.com/user-attachments/files/17615638/transcript_tpm_ctype.csv)
[transcript_count_ctype.csv](https://github.com/user-attachments/files/17615639/transcript_count_ctype.csv)

**Comment by kvittingseerup** [comment]
There is boy only one isoform in your data. But one isoform is in the annotation but not quantified. 

How was the data quantified?

**Comment by MetteBoge** [comment]
I am not completely sure what you mean. The data is from  the Xena browser (TCGA PANCAN): 
https://xenabrowser.net/datapages/?dataset=tcga_rsem_isoform_tpm&host=https%3A%2F%2Ftoil.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443 
https://xenabrowser.net/datapages/?dataset=tcga_expected_count&host=https%3A%2F%2Ftoil.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443

I have just made a subset of this data containing isoforms from 14 target genes and in this case patients with Pancreatic Cancer.

**Comment by chunxubioinfor** [comment]
Ahh I misunderstood the warning message. The warning means that the isoform_id of your count matrix and gtf file are slightly different and there is one isoform in the gtf file but not show up in the count matrix.

**Comment by MetteBoge** [comment]
Ah okay I understand now. The warning is gone now, but I just dont find any "estimated_genes_with_dtu". Thank you!

**Comment by chunxubioinfor** [comment]
You could continue to do filtration and apply SatuRn or Dexseq to test for isoform switch, but I have to say there's a chance that can't get any positive results as you only focused on 14 genes. Maybe try expanding the target genes list?

**Comment by kvittingseerup** [comment]
That is not unexpected since you only test a few genes and the estimate uses a subset of the data 😊

If you run into more problems just make another GitHub issue 😊
________________________________
Fra: MetteBoge ***@***.***>
Sendt: Monday, November 4, 2024 11:36:36 AM
Til: kvittingseerup/IsoformSwitchAnalyzeR ***@***.***>
Cc: Kristoffer Vitting-Seerup ***@***.***>; Comment ***@***.***>
Emne: Re: [kvittingseerup/IsoformSwitchAnalyzeR] 1 isoforms were only found in the annotation (Issue #257)


Ah okay I understand now. The warning is gone now, but I just dont find any "estimated_genes_with_dtu". Thank you!

**Comment by kvittingseerup** [comment]
> Just checked your count matrix, and I found that ENST00000582017.1 is in the annotation but not in these 64 isoforms you got. Can you check if there's something wrong when you did the subset? :)

This can also be due to the quantification tool used - some tools remove non-unique transcipts

**Comment by MetteBoge** [comment]
Thank you! I will keep investigating for isoform switches. I am only interested in the 14 target genes so unless the algorithm borrow information across genes, I will just use the subset. 
Thank you
